test/index.test-d.ts

```diff
@@ -67,7 +67,7 @@ conf.set({
 	}
 });
 
-expectType<string | undefined>(conf.get('foo'));
+expectType<string>(conf.get('foo'));
```

Comment:
The second one was indeed incorrect. But Is it correct to assume that a value will never be undefined when we do not pass a default? I think we should still make it possible to be undefined to encourage people to handle an undefined state.

Reply:
Yes, value will never be undefined unless developer make a bug such in this test, type `UnicornFoo` is source of truth.
```ts
type UnicornFoo = {
	foo: string;
};
const conf = new Conf<UnicornFoo>({accessPropertiesByDotNotation: true});
```
Maybe we should validate `default` option, but it is out of scope of this PR.
Like this
```ts
	defaults?: Readonly<{
		[P in keyof T]: Exclude<T[P], undefined>
	}>;
```

Reply:
> Maybe we should validate default option, but it is out of scope of this PR.

That would be nice, yes, but I agree, it should be a separate PR.